components/lvgl_esp32_drivers/lvgl_tft/Kconfig

```diff
@@ -294,6 +294,39 @@ menu "LVGL TFT Display controller"
             select LV_TFT_DISPLAY_PROTOCOL_SPI
     endchoice
 
+    # TODO: This configuration option will be moved into the LVGL
+    # Kconfig menuconfig, let's leave that for the near future.
+    config LV_COLOR_16_SWAP
+        bool
+        default y if LV_TFT_DISPLAY_CONTROLLER_ILI9341
+        default n if LV_TFT_DISPLAY_CONTROLLER_ILI9488
+        default y if LV_TFT_DISPLAY_CONTROLLER_ST7789
+        default y if LV_TFT_DISPLAY_CONTROLLER_ST7735S
+        default y if LV_TFT_DISPLAY_CONTROLLER_HX8357
+        default n if LV_TFT_DISPLAY_CONTROLLER_SH1107
+        default n if LV_TFT_DISPLAY_CONTROLLER_SSD1306
+        default y if LV_TFT_DISPLAY_CONTROLLER_ILI9486
+        default n if LV_TFT_DISPLAY_CONTROLLER_FT81X
+        default y if LV_TFT_DISPLAY_CONTROLLER_RA8875
+        default y if LV_TFT_DISPLAY_CONTROLLER_GC9A01
+    
+    # TODO: This configuration option will be moved into the LVGL
+    # Kconfig menuconfig, let's leave that for the near future.
+    config LV_COLOR_DEPTH
+        # Color depth:
+        # 1: 1 byte per pixel
+        # 8: RGB233
+        # 16: RGB565
+        # 32: ARGB8888
+        int
+
+        default 1 if LV_TFT_DISPLAY_USER_CONTROLLER_SH1107
+        default 1 if LV_TFT_DISPLAY_USER_CONTROLLER_SSD1306
+        default 1 if LV_TFT_DISPLAY_USER_CONTROLLER_IL3820
+        default 1 if LV_TFT_DISPLAY_USER_CONTROLLER_JD79653A
+        default 1 if LV_TFT_DISPLAY_USER_CONTROLLER_UC8151D
+        default 16
```

Comment:
I'm not a Kconfig expert, but does having the `default 16` after every other default cause it to override them?

Reply:
May be, I built a project with an SSD1306 controller selected and got the symbol defined as 1. I will try to investigate further.

Reply:
What if that option is moved into the lvgl Kconfig, like so:

```
    choice LV_COLOR_DEPTH
        prompt "Color depth."
        default LV_COLOR_DEPTH_16
        help
            Color depth to be used.

        config LV_COLOR_DEPTH_32
            bool "32: ARGB8888"
        config LV_COLOR_DEPTH_16
            bool "16: RGB565"
        config LV_COLOR_DEPTH_8
            bool "8: RGB232"
        config LV_COLOR_DEPTH_1
            bool "1: 1 byte per pixel"
    endchoice
```

in lvgl Kconfig:

```
#ifdef (CONFIG_LV_COLOR_DEPTH_1)
#define LV_COLOR_DEPTH     0
#elif defined (CONFIG_LV_COLOR_DEPTH_8)
#define LV_COLOR_DEPTH     8
#elif defined (CONFIG_LV_COLOR_DEPTH_16)
#define LV_COLOR_DEPTH     16
#elif defined (CONFIG_LV_COLOR_DEPTH_32)
#define LV_COLOR_DEPTH     32
#endif
```

"Problem" is, the user will need to choose the color depth in lvgl menuconfig, it will not be handled automatically. I tried using a select when choosing the diplay, but I got the following warning:
```
warning: the choice symbol LV_COLOR_DEPTH_16 (defined at /home/user/lv_port_esp32/components/lvgl/Kconfig:56) is selected by the following symbols, but select/imply has no effect on choice symbols
 - LV_TFT_DISPLAY_USER_CONTROLLER_ILI9341 (defined at /home/user/lv_port_esp32/components/lvgl_esp32_drivers/lvgl_tft/Kconfig:230)
 - LV_TFT_DISPLAY_USER_CONTROLLER_ILI9481 (defined at /home/user/lv_port_esp32/components/lvgl_esp32_drivers/lvgl_tft/Kconfig:235)
 - LV_TFT_DISPLAY_USER_CONTROLLER_ILI9486 (defined at /home/user/lv_port_esp32/components/lvgl_esp32_drivers/lvgl_tft/Kconfig:240)
 - LV_TFT_DISPLAY_USER_CONTROLLER_ILI9488 (defined at /home/user/lv_port_esp32/components/lvgl_esp32_drivers/lvgl_tft/Kconfig:245)
 - LV_TFT_DISPLAY_USER_CONTROLLER_ST7789 (defined at /home/user/lv_port_esp32/components/lvgl_esp32_drivers/lvgl_tft/Kconfig:250)
 - LV_TFT_DISPLAY_USER_CONTROLLER_GC9A01 (defined at /home/user/lv_port_esp32/components/lvgl_esp32_drivers/lvgl_tft/Kconfig:255)
 - LV_TFT_DISPLAY_USER_CONTROLLER_ST7735S (defined at /home/user/lv_port_esp32/components/lvgl_esp32_drivers/lvgl_tft/Kconfig:260)
 - LV_TFT_DISPLAY_USER_CONTROLLER_HX8357 (defined at /home/user/lv_port_esp32/components/lvgl_esp32_drivers/lvgl_tft/Kconfig:265)
 - LV_TFT_DISPLAY_USER_CONTROLLER_FT81X (defined at /home/user/lv_port_esp32/components/lvgl_esp32_drivers/lvgl_tft/Kconfig:282)
 - LV_TFT_DISPLAY_USER_CONTROLLER_RA8875 (defined at /home/user/lv_port_esp32/components/lvgl_esp32_drivers/lvgl_tft/Kconfig:305)
warning: the choice symbol LV_COLOR_DEPTH_1 (defined at /home/user/lv_port_esp32/components/lvgl/Kconfig:60) is selected by the following symbols, but select/imply has no effect on choice symbols
 - LV_TFT_DISPLAY_USER_CONTROLLER_SH1107 (defined at /home/user/lv_port_esp32/components/lvgl_esp32_drivers/lvgl_tft/Kconfig:270)
 - LV_TFT_DISPLAY_USER_CONTROLLER_SSD1306 (defined at /home/user/lv_port_esp32/components/lvgl_esp32_drivers/lvgl_tft/Kconfig:276)
 - LV_TFT_DISPLAY_USER_CONTROLLER_IL3820 (defined at /home/user/lv_port_esp32/components/lvgl_esp32_drivers/lvgl_tft/Kconfig:287)
 - LV_TFT_DISPLAY_USER_CONTROLLER_JD79653A (defined at /home/user/lv_port_esp32/components/lvgl_esp32_drivers/lvgl_tft/Kconfig:293)
 - LV_TFT_DISPLAY_USER_CONTROLLER_UC8151D (defined at /home/user/lv_port_esp32/components/lvgl_esp32_drivers/lvgl_tft/Kconfig:299)
```

What do you think @embeddedt?

Reply:
I don't have a problem with the way you have it right now; I just wanted to clarify that it works the way it's intended to. Personally I prefer the integer selection over a choice selection.

Reply:
I see, I don't have problem neighter, but this change will end up in the main lvgl repo, so with the last comment I wanted to start moving all lvgl configuration options into the lvgl Kconfig.